src/staticarrays_support.jl

```diff
@@ -0,0 +1,10 @@
+import StaticArrays: StaticArray, tuple_prod
+
```

Comment:
Maybe we could also add some docs here (I think just for `staticschema` it should be enough) to mention that the `staticschema` of a `StaticArray` is the `staticschema` of the underlying flat tuple.

Reply:
Done. I was going to add a docstring for expected output, but the output looks like that of the old nested storage?
```julia
julia> StructArrays.staticschema(SVector{2})
NamedTuple{(:data,), Tuple{Tuple{T, T} where T}}
```

Reply:
Ah, I see. In practice what happens is that the subtyping fails because it cannot determine the parameter `T`  (as you did not specify it in the type `SVector{2}`).

I think that's ok, because it's not a concrete type: an actual vector can never be `SVector{2}`, it would be for example `SVector{2, Float64}`. We should probably add a test that `staticschema(SVector{2, Float64})` (or `typeof(s)` for `s` a static array) does the correct thing. I should also double check that `staticschema` is only ever called with concrete types.

Reply:
Thanks for the explanation! Adding the test for `staticschema`. 

> I should also double check that staticschema is only ever called with concrete types.

Should this be added to the README or docs for `staticschema`?

Reply:
Yes good point, the docstring should say that it is only guaranteed to work on concrete types.

Reply:
I'm happy to add that, or I can leave that to you.

Reply:
I think this was resolved apart from the note about `staticschema` and concrete types. The docstring has a concrete type example in it, so resolving convo for now.